### PR TITLE
Filter clause doc

### PR DESCRIPTION
**Pull request by Alex-Monahan**
I documented the FILTER clause with some verbiage, some examples, and an updated railroad diagram. I could use some help on additional use cases where we should advise folks to use it. I haven't seen a performance advantage over the CASE WHEN approach, so there are only a few situations where I would recommend the FILTER clause. 

Thanks!

### CODE REVIEW

**Comment by Alex-Monahan** [comment]
Richard gave me some good advice, so I'm still revising this! I hope to finish it in the next day or two with an example that uses the "FIRST" function.

File: docs/sql/query_syntax/filter.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+---
+layout: docu
+title: FILTER Clause
+selected: Documentation/SQL/Query Syntax/Filter
+expanded: SQL
+railroad: query_syntax/filter.js
+---
+The `FILTER` clause may optionally follow an aggregate function in a `SELECT` statement. This will filter the rows of data that are fed into the aggregate function in the same way that a `WHERE` clause filters rows, but localized to the specific aggregate function. `FILTER`s are not currently able to be used when the aggregate function is in a windowing context. 
+
+There are multiple types of situations where this is useful, including when evaluating multiple aggregates with different filters, and when creating a pivoted view of a dataset. `FILTER` provides a cleaner syntax for pivoting data when compared with the more traditional `CASE WHEN` approach discussed below. Some aggregate functions also do not filter out null values, so using a `FILTER` clause will return valid results when at times the `CASE WHEN` approach will not.
```
Comment (hawkfish) [inline docs/sql/query_syntax/filter.md:10]: Specifically, `FIRST`, `LAST` and `LIST`/`ARRAY_AGG`.

File: _data/menu.json [hunk 2]
```diff
@@ -225,6 +225,10 @@
                     {
                       "page": "VALUES",
                       "url": "/docs/sql/query_syntax/values"
+                    },
```
Comment (hawkfish) [inline _data/menu.json:228]: The rest of the menu items appear to be in alphabetical order?

File: js/query_syntax/filter.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+					Optional(Keyword("DISTINCT"), "skip"),
+					OneOrMore(Sequence([
+						Expression()
+					]), Keyword(",")),
+					Optional(Expandable("order-by", options, "order-by", GenerateAggregateOrderBy))
+				]),
+				new Skip()
+			]),
+			Keyword(")"),
+			Optional(Expandable("filter-clause",options,"filter-clause",GenerateFilterClause))
```
Comment (hawkfish) [inline js/query_syntax/filter.js:35]: Does this need to be expandable? There is plenty of room.

File: js/expressions/aggregate.js [hunk 4]
```diff
@@ -6,9 +6,19 @@ function GenerateAggregateOrderBy(options = {}) {
 	]
 }
 
+function GenerateFilterClause(options = {}) {
```
Comment (hawkfish) [inline js/expressions/aggregate.js:9]: Can this function be shared?

File: js/expressions/aggregate.js [hunk 5]
```diff
@@ -21,7 +31,8 @@ function GenerateAggregate(options = {}) {
 				]),
 				new Skip()
 			]),
-			Keyword(")")
+			Keyword(")"),
+			Optional(Expandable("filter-clause",options,"filter-clause",GenerateFilterClause)),
```
Comment (hawkfish) [inline js/expressions/aggregate.js:35]: Does this need to be expandable?

**Review by hawkfish — APPROVED** [review]

**Comment by Mytherin** [comment]
@hawkfish is this ready to merge, or should @Alex-Monahan address your requests first?

**Comment by Alex-Monahan** [comment]
I have some uncommitted changes that help address some of those comments! I'll push them today or tomorrow and comment here when it's ready from my side. Thanks Mark!

File: _data/menu.json (hunk 2 above)
Comment (Alex-Monahan) [reply _data/menu.json:228]: The ordering isn't quite alphabetical. It appears to be roughly in importance order (which is why I put FILTER at the bottom!). Do you mind if I leave it where it is?

File: docs/sql/query_syntax/filter.md (hunk 1 above)
Comment (Alex-Monahan) [reply docs/sql/query_syntax/filter.md]: Added!

File: js/query_syntax/filter.js (hunk 3 above)
Comment (Alex-Monahan) [reply js/query_syntax/filter.js:35]: Fixed!

File: js/expressions/aggregate.js (hunk 4 above)
Comment (Alex-Monahan) [reply js/expressions/aggregate.js:9]: Fixed!

File: js/expressions/aggregate.js (hunk 5 above)
Comment (Alex-Monahan) [reply js/expressions/aggregate.js:35]: Fixed!

**Comment by Alex-Monahan** [comment]
Ok, changes made and feedback addressed! Ready to merge @Mytherin. Thanks for taking a look at this @hawkfish - much appreciated.

**Pull request merged**